proxy/proxy.go

```diff
@@ -717,6 +722,40 @@ func (c *client) handleQuery(raw *frame.RawFrame, msg *codecs.PartialQuery, body
 	}
 }
 
+func (c *client) handleRequestId(body *frame.Body, frm *frame.RawFrame) ([]byte, *frame.RawFrame) {
+	if !c.proxy.config.EnableTracing {
+		return nil, frm
+	}
+	if body.CustomPayload == nil {
+		frm.Header.Flags = frm.Header.Flags.Add(primitive.HeaderFlagCustomPayload)
+		body.CustomPayload = make(map[string][]byte)
+	}
+	var reqId []byte
+	if id, ok := body.CustomPayload["request-id"]; !ok {
+		cid, err := uuid.New().MarshalBinary()
+		if err != nil {
+			return nil, frm
+		}
+		body.CustomPayload["request-id"] = cid
+		reqId = cid
+		var buffer bytes.Buffer
+		writer := bufio.NewWriter(&buffer)
+		err = c.codec.EncodeBody(frm.Header, body, writer)
+		if err != nil {
+			return nil, frm
+		}
+		err = writer.Flush()
+		if err != nil {
+			return nil, frm
+		}
+		frm.Body = buffer.Bytes()
+	} else {
+		reqId = id
+	}
+	c.proxy.logger.Info("received request id", zap.String("idHex", hex.EncodeToString(reqId)))
```

Comment:
"received request id" seems a bit confusing because it might be generated in the router. Should it be differentiated?